File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.java
class Solution {
   public  int lengthOfLongestSubstring(String s) {
        return find("",s,0);
        
    }

    public  int find(String p,String s,int max) {
        max=Math.max(p.length(),max);
        if(s.isEmpty())
        {
            return max;
        }
        char ch=s.charAt(0);
        if(p.length()==0)
        {
            return find(p+ch,s.substring(1),max);
        }
        for (int i=0;i<p.length();i++) {
            if(p.charAt(i)==ch){
                return find(p.substring(i+1,p.length()),s,max);
            }
        }
        return find(p+ch,s.substring(1),max);
    }
}

File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        unordered_map<char,int> mp;
    int i=0;
    int j=0;
    int ans=0;
    while(j<s.size()){
         if(mp[s[j]]==0){
            mp[s[j]]=j+1;
         }
         else{
            while(i<=mp[s[j]]){
                mp[s[i]]=0;
                i++;
            }
            mp[s[j]]=j+1;
         }
         ans=max(j-i+1,ans);
         j++;
    }
        return ans;
    }
};